src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsTransportTest.php

```diff
@@ -11,58 +11,125 @@
+    public function testItCanRejectAMessageViaTheReceiver(): void
+    {
+        $envelope = new Envelope(new \stdClass());
+        $this->receiver->expects($this->once())->method('reject');
+        $this->transport->reject($envelope);
+    }
 
-        $serializer->method('decode')->with(['body' => 'body', 'headers' => ['my' => 'header']])->willReturn(new Envelope($decodedMessage));
-        $connection->method('get')->willReturn($sqsEnvelope);
+    public function testItCanGetMessageCountViaTheReceiver(): void
+    {
+        $messageCount = 15;
+        $this->receiver->expects($this->once())->method('getMessageCount')->willReturn($messageCount);
+        $this->assertSame($messageCount, $this->transport->getMessageCount());
+    }
 
-        $envelopes = iterator_to_array($transport->get());
-        $this->assertSame($decodedMessage, $envelopes[0]->getMessage());
+    public function testItCanSendAMessageViaTheSender(): void
+    {
+        $envelope = new Envelope(new \stdClass());
+        $this->sender->expects($this->once())->method('send')->willReturn($envelope);
+        $this->assertSame($envelope, $this->transport->send($envelope));
     }
 
-    public function testTransportIsAMessageCountAware()
+    public function testItCanSetUpTheConnection(): void
     {
-        $transport = $this->getTransport();
+        $this->connection->expects($this->once())->method('setup');
+        $this->transport->setup();
+    }
+
+    public function testItConvertsHttpExceptionDuringSetupIntoTransportException(): void
+    {
+        $this->connection
+            ->expects($this->once())
+            ->method('setup')
+            ->willThrowException($this->createHttpException());
 
-        $this->assertInstanceOf(MessageCountAwareInterface::class, $transport);
+        $this->expectException(TransportException::class);
+
+        $this->transport->setup();
     }
 
-    private function getTransport(SerializerInterface $serializer = null, Connection $connection = null)
+    public function testItCanResetTheConnection(): void
     {
-        $serializer = $serializer ?: $this->getMockBuilder(SerializerInterface::class)->getMock();
-        $connection = $connection ?: $this->getMockBuilder(Connection::class)->disableOriginalConstructor()->getMock();
+        $this->connection->expects($this->once())->method('reset');
+        $this->transport->reset();
+    }
 
-        return new AmazonSqsTransport($connection, $serializer);
+    public function testItConvertsHttpExceptionDuringResetIntoTransportException(): void
+    {
+        $this->connection
+            ->expects($this->once())
+            ->method('reset')
+            ->willThrowException($this->createHttpException());
+
+        $this->expectException(TransportException::class);
+
+        $this->transport->reset();
+    }
+
+    private function createHttpException(): HttpException
+    {
+        return new ServerException($this->createMock(ResponseInterface::class));
     }
 }
```

Comment:
Please revert any refactoring and only add new tests. It causes merge conflicts and make the diff harder to review

Reply:
Sure, I'll restore these methods. I removed them because I found them redundant but maybe I'm wrong, or maybe it's just a topic for another PR?